File: _124나라의숫자.java
class _124나라의숫자 {
    int mod124[] = {4, 1, 2};
    public String solution(int n) {
        StringBuilder answer = new StringBuilder();
        while(n!=0){
            int mod = n%3;
            n/=3;
            if(mod==0) n--;
            answer.append((char)('0'+mod124[mod]));
        }
        
        answer.reverse();
        
        return answer.toString();
    }
}

File: 프로그래머스/124나라의숫자.cpp
#include <string>
#include <vector>

using namespace std;

string solution(int n) {
    string answer = "";
    int div[4]={0,1,2,4};
    int rest[3]={4, 1, 2};

    int r;
    vector<int> v;
    while(n){
        r=n%3;
        v.push_back(rest[r]);
        n/=3;
        if(r==0) n--;
    }
    for(int i=v.size()-1; i>=0; i--)
        answer+='0'+(char)v[i];


    return answer;
}
